lib/matplotlib/legend.py

```diff
@@ -1269,6 +1278,44 @@ def _find_best_position(self, width, height, renderer, consider=None):
     def contains(self, event):
         return self.legendPatch.contains(event)
 
+    def set_draggable(self, state, use_blit=False, update='loc'):
+        """
+        Enable or disable mouse dragging support of the legend.
+
+        Parameters
+        ----------
+        state : bool
+            ``True`` / ``False`` enables / disables mouse dragging.
+        use_blit : bool, optional
+            Blitting is an old and basic but fast composition method for
+            images. Usually, you won't need to set this. However, you may use
```

Comment:
I don't think blitting should reduce the quality in any way, it is just doing a bit more book keeping to be lazy!

Reply:
Ok, no idea what you are actually doing behind the scenes here.

Both [FuncAnimation](https://matplotlib.org/api/animation_api.html#funcanimation) and the linked wikipedia article, give the impression that this is not something one would want to use:
- "old technique"
- "Modern graphics software has almost completely replaced bitwise operations with more general mathematical operations used for effects such as alpha compositing. This is because bitwise operations on color displays do not usually produce results that resemble the physical combination of lights or inks."

From a user point of view, I'm really lost here. I've been trying to make some reason of the option, but it's unclear what this means for my plot other than making dragging faster. If that'd be all, we could enable it by default and throw away the option.

Reply:
"The old ways are best"  ;)   The "old technique" comment is supposed to be re-assuring that this is a good thing to do, not a warning not to use it.   This rough method is heavily used in web browsers and modern GUI toolkits (do not bother to re-render pixels that have not changed). 

The reason you want to use it is for speed, but the cost is a lot more book keeping and not all of the backends support it.

We can not just drop the option (because that would be a pretty big API break for anyone who is currently using it).

Reply:
Thanks for the explanation. I don't want to break the API. Saying, we could throw away the option was just my clumsy way of expressing, that I don't understand its purpose. From what I've understood so far, there's no reason not to use `use_blit=True`. Since you chose to add the option some time in the past instead of just always trying to use blit, obviously I'm missing something.

Because I'm stumbling here, I assume that other users will have the same problem. So I'd like to explain the option thoroughly in the docstring. Therefore I need to know:

1. What is disadvantage is for a user? When would a user *not* want blitting.

The book keeping is done by the underlying code. So as a user that's nothing I have to care about, is it? Maybe a bit more memory usage? But that's probably negligible. Or are there maybe situations where the book keeping is getting wrong, resulting in artifacts in the plot?

2. Is there a good reason that `use_blit` is False by default or is it historic?

IMO "Not supported in all backends" is no reason. `DraggableBase` checks if the canvas supports blit. So
`use_blit` is already effectively "use blit if supported by the backend".

Reply:
I think the old OSX backend used to blow up if you tried to do any blitting with it, I am not 100% sure that it gracefully decays on all current backends (and that all backends correctly report if they support blitting).  

I suspect that when this feature was added 'off by default' was the nod to back compatibility.  If all backends do gracefully decay there should be no downside to setting this to True.  If the user is doing their own blitting, there may be issues where the legend will show up in few places due to the two sets of book-keeping getting out of sync.